botcore/utils/messages.py

```diff
@@ -0,0 +1,306 @@
+import asyncio
+import random
+import re
+from functools import partial
+from io import BytesIO
+from typing import Callable, Sequence
+
+import discord
+from discord.ext import commands
+
+from botcore.utils import scheduling
+from botcore.utils.logging import get_logger
+
+
+log = get_logger(__name__)
+
+
+def reaction_check(
+    reaction: discord.Reaction,
+    user: discord.abc.User,
+    *,
+    message_id: int,
+    allowed_emoji: Sequence[str],
+    allowed_users: Sequence[int],
+    mod_roles: Sequence[int],
+    allow_mods: bool = True,
```

Comment:
I've only now seen this PR. I think my [original suggestion](https://canary.discord.com/channels/267624335836053506/635950537262759947/1023309162865770647) would simplify this quite a bit, where these args are replaced with an `allowed_roles` Sequence. Similar to how https://github.com/python-discord/bot-core/blob/main/botcore/utils/interactions.py#L28 works.

Same comment for `reaction_check` below.

Reply:
Yeah, I was thinking of having `allowed_roles` since there's already `allowed_users`. I think it makes sense to swap to this.

Reply:
Implemented in [ed8b4f7](https://github.com/python-discord/bot-core/pull/141/commits/ed8b4f787a251227a1da3572459c5722c3f99133) :+1: